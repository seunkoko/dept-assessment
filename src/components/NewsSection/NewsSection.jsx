// eslint-disable-next-line
import React from 'react';

import ImageContainer from '../ImageContainer/ImageContainer';

import './NewsSection.scss';

const NewsSection = ({ data }) => {
	return (
    <>
      <div className="news-style">
        {
          data.map(news => {
            const newsToDisplay = <div
              className="news-item"
              key={news.id}
              >
                <ImageContainer
                  data={news}
                />
              </div>
            return newsToDisplay;
          })
        }
      </div>
    </>
	);
};

export default NewsSection;
