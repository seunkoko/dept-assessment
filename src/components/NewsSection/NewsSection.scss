@import "../../stylesheets/variables";

.news-style {
  margin: 30px 0;
  width: 100%;
  display: grid;
  grid-template-columns: 49% 49%;
  grid-column-gap: 2%;
  grid-row-gap: 30px;

  @media (max-width: $phone-width) {
    display: flex;
    flex-direction: column;
    margin-bottom: 0;
  }
}

.news-item {

  > div {
    > img {
      height: 400px;
      width: 100%;

      @media (max-width: $phone-width) {
        height: 250px;
      }
    }
  }
  
  @media (max-width: $phone-width) {
    margin-bottom: 30px;
  }
}