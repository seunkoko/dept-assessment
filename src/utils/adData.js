import florensis from '../assets/images/florensis.png';

export const imageData = {
  id: 'adImage',
  image: florensis,
  title: 'Florensis',
  description: 'Rethinking the online ecosystem',
  moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
}

export const topFirstNews = {
  id: 'topFirstNews',
  title: 'MICROSOFT',
  news: 'Tapping into Ireland’s unique gaming culture and bringing a fresh flavour to their Xbox social media channels',
  moreDescription: ''
}

export const topSecondNews = {
  id: 'topSecondNews',
  title: 'O\'NEIL',
  news: 'Integrating existing content into O’Neill’s new e-commerce platform',
  moreDescription: ''
}

export const bottomFirstNews = {
  id: 'bottomFirstNews',
  title: 'BUTLINS',
  news: 'Enhancing customer experience for personalised holiday planning using real time data',
  moreDescription: ''
}

export const bottomSecondNews = {
  id: 'bottomSecondNews',
  title: 'VACANSESELECT',
  news: 'Predicting booking behavior for holidays with smart data, voice search and machine learning',
  moreDescription: ''
}
