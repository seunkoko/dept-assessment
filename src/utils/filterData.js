
export const workFilter = [
  {
    id: 1,
    name: 'Frontend Engineer',
  }, 
  {
    id: 2,
    name: 'Backend Engineer',
  }, 
  {
    id: 3,
    name: 'Data Scientist',
  },
];

export const industryFilter = [
  {
    id: 1,
    name: 'E-Commerce',
  }, 
  {
    id: 2,
    name: 'Medicine',
  }, 
  {
    id: 3,
    name: 'Robotics',
  },
];