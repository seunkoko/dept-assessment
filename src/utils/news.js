import bolcom from '../assets/images/bolcom.png';
import kempen from '../assets/images/kempen.png';
import philips from '../assets/images/philips.png';
import gemeentemuseum from '../assets/images/gemeentemuseum.png';
import lightning from '../assets/images/lightning.png';
import tui from '../assets/images/tui.png';
import chocomel from '../assets/images/chocomel.png';
import zalando from '../assets/images/zalando.png';
import jbl from '../assets/images/jbl.png';
import konin from '../assets/images/koninklijke-bibliotheek.png';
import liberty from '../assets/images/liberty-global.png';
import arla from '../assets/images/arla.png';

export const firstNewsData = [
  {
    id: 'bolcom',
    image: bolcom,
    title: 'BOLCOM',
    description: 'A summer island in the Netherlands',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'kempen',
    image: kempen,
    title: 'KEMPEN',
    description: 'Not some average banking website',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'phillips',
    image: philips,
    title: 'PHILIPS',
    description: 'Beautiful design meets innovative technology',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'gemeentemuseum',
    image: gemeentemuseum,
    title: 'GEMEENTEMUSEUM',
    description: 'A hundred years of Mondriaan & De Stijl',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]

export const secondNewsData = [
  {
    id: 'lightning',
    image: lightning,
    title: 'BE LIGHTNING',
    description: 'Delivering clarity on a global scale',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'tui',
    image: tui,
    title: 'TUI',
    description: 'Swipe to find your next holiday destination',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]

export const thirdNewsData = [
  {
    id: 'chocomel',
    image: chocomel,
    title: 'CHOCOMEL',
    description: 'A campaign for the limited edition letter packs',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'jbl',
    image: jbl,
    title: 'JBL',
    description: 'Live like a champion with Jerome Booteng',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'zalando',
    image: zalando,
    title: 'ZALANDO',
    description: 'Innovative SEO and content strategy for Zalando',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'konin',
    image: konin,
    title: 'KONINKLIJKE BIBLIOTHEEK',
    description: 'The search for the most influential book ever',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]

export const fourthNewsData = [
  {
    id: 'liberty',
    image: liberty,
    title: 'LIBERTY GLOBAL',
    description: 'Delivering complex commerce solutions',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
  {
    id: 'arla',
    image: arla,
    title: 'ARLA',
    description: 'Swipe to find your next holiday destination',
    moreDescription: 'Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.'
  },
]
