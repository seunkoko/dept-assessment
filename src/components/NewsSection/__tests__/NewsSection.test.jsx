import React from 'react';

import { shallow } from 'enzyme';

import NewsSection from '../NewsSection';

import { firstNewsData } from '../../../utils/news';

describe('NewsSection', () => {
	const component = shallow(<NewsSection data={firstNewsData} />);

	it('should render NewsSection component', () => {
		expect(component).toMatchSnapshot();
  });
  
});