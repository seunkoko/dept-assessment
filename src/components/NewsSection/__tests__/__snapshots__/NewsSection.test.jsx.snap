// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSection should render NewsSection component 1`] = `
<Fragment>
  <div
    className="news-style"
  >
    <div
      className="news-item"
      key="bolcom"
    >
      <ImageContainer
        data={
          Object {
            "description": "A summer island in the Netherlands",
            "id": "bolcom",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "BOLCOM",
          }
        }
      />
    </div>
    <div
      className="news-item"
      key="kempen"
    >
      <ImageContainer
        data={
          Object {
            "description": "Not some average banking website",
            "id": "kempen",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "KEMPEN",
          }
        }
      />
    </div>
    <div
      className="news-item"
      key="phillips"
    >
      <ImageContainer
        data={
          Object {
            "description": "Beautiful design meets innovative technology",
            "id": "phillips",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "PHILIPS",
          }
        }
      />
    </div>
    <div
      className="news-item"
      key="gemeentemuseum"
    >
      <ImageContainer
        data={
          Object {
            "description": "A hundred years of Mondriaan & De Stijl",
            "id": "gemeentemuseum",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "GEMEENTEMUSEUM",
          }
        }
      />
    </div>
  </div>
</Fragment>
`;
